libethereum/State.h

```diff
@@ -169,200 +197,250 @@ using ChangeLog = std::vector<Change>;
+                                  SealEngineFace const& _sealEngine);
+
+    /// Check if the address is in use.
+    bool addressInUse(Address const& _address) const;
+
+    /// Check if the account exists in the state and is non empty (nonce > 0 ||
+    /// balance > 0 || code nonempty). These two notions are equivalent after
+    /// EIP158.
+    bool accountNonemptyAndExisting(Address const& _address) const;
+
+    /// Check if the address contains executable code.
+    bool addressHasCode(Address const& _address) const;
+
+    /// Get an account's balance.
+    /// @returns 0 if the address has never been used.
+    u256 balance(Address const& _id) const;
+
+    /// Add some amount to balance.
+    /// Will initialise the address if it has never been used.
+    void addBalance(Address const& _id, u256 const& _amount);
+
+    /// Subtract the @p _value amount from the balance of @p _addr account.
+    /// @throws NotEnoughCash if the balance of the account is less than the
+    /// amount to be subtrackted (also in case the account does not exist).
+    void subBalance(Address const& _addr, u256 const& _value);
+
+    /**
+     * @brief Transfers "the balance @a _value between two accounts.
+     * @param _from Account from which @a _value will be deducted.
+     * @param _to Account to which @a _value will be added.
+     * @param _value Amount to be transferred.
+     */
+    void transferBalance(Address const& _from,
+                         Address const& _to,
+                         u256 const& _value)
+    {
+        subBalance(_from, _value);
+        addBalance(_to, _value);
+    }
+
+    /// Get the root of the storage of an account.
+    h256 storageRoot(Address const& _contract) const;
+
+    /// Get the value of a storage position of an account.
+    /// @returns 0 if no account exists at that address.
+    u256 storage(Address const& _contract, u256 const& _memory) const;
+
+    /// Set the value of a storage position of an account.
+    void setStorage(Address const& _contract,
+                    u256 const& _location,
+                    u256 const& _value);
+
+    /// Clear the storage root hash of an account to the hash of the empty trie.
+    void clearStorage(Address const& _contract);
+
+    /// Create a contract at the given address (with unset code and unchanged
+    /// balance).
+    void createContract(Address const& _address);
+
+    /// Sets the code of the account. Must only be called during / after
+    /// contract creation.
+    void setCode(Address const& _address, bytes&& _code);
+
+    /// Delete an account (used for processing suicides).
+    void kill(Address _a);
+
+    /// Get the storage of an account.
+    /// @note This is expensive. Don't use it unless you need to.
+    /// @returns map of hashed keys to key-value pairs or empty map if no
+    /// account exists at that address.
+    std::map<h256, std::pair<u256, u256>> storage(
```

Comment:
I find this line ending with `(` a bit scary.  I think I can get used to it.

Reply:
The only tradeoff we can have here is to allow putting the return type on its own line... let me try this.